app.py

```
@@ -143,24 +167,30 @@ def on_exit():
                 relief="flat",
             )
             exit_button.pack(side="right", padx=20)
-
             popup.mainloop()
 
         except Exception as e:
             messagebox.showerror("Error", f"Failed to load image: {e}")
 
     def start_popup(self):
         while True:
+            logging.info(f"Waiting {self.interval} seconds before showing the next popup.")
```

Ruff

Logging statement uses f-string

(G004)